baseapp/abci.go

```diff
@@ -231,9 +232,18 @@ func (app *BaseApp) Commit() (res abci.ResponseCommit) {
 	app.deliverState = nil
 
 	defer func() {
-		if app.haltHeight > 0 && uint64(header.Height) == app.haltHeight {
-			app.logger.Info("halting node per configuration", "height", app.haltHeight)
-			os.Exit(0)
+		var halt bool
+
+		switch {
+		case app.haltHeight > 0 && uint64(header.Height) == app.haltHeight:
+			halt = true
+
+		case app.haltTime > 0 && header.Time.UnixNano() >= int64(app.haltTime):
+			halt = true
+		}
+
+		if halt {
+			app.halt()
```

Comment:
So the node will gracefully halt here. However, there is nothing stopping an operator from restarting their node and continuing to process and commit blocks. We could add a complimentary check in `BeginBlock` to prohibit progression. Thoughts @zmanian?

Reply:
Nvm, maybe we should move this entire logic to BeginBlock.

Reply:
I agree to your assessment, IMHO this should live in `BeginBlock`